File: data/src/main/java/com/github/anrimian/musicplayer/data/storage/files/FileManager.java
package com.github.anrimian.musicplayer.data.storage.files;

import java.io.File;

public class FileManager {

    public static void deleteFile(String path) {
        File file = new File(path);
        if (!file.exists()) {
            return;
        }
        //noinspection ResultOfMethodCallIgnored
        file.delete();
    }


    public static void deleteEmptyDirectory(File fileOrDirectory) {
        if (fileOrDirectory.isDirectory()) {
            File[] files = fileOrDirectory.listFiles();
            if (files == null) {
                return;
            }
            for (File child : files) {
                if (fileOrDirectory.isDirectory()) {
                    deleteEmptyDirectory(child);
                }
            }
            if (files.length == 0) {
                //noinspection ResultOfMethodCallIgnored
                fileOrDirectory.delete();
            }
        }
    }

    //TODO think about this
//    public void deleteEmptyDirectory(File fileOrDirectory) {
//        if (fileOrDirectory.isDirectory()) {
//            File[] files = fileOrDirectory.listFiles();
//            if (files != null) {
//                for (File child : files) {
//                    if (fileOrDirectory.isDirectory()) {
//                        deleteEmptyDirectory(child);
//                    }
//                }
//            }
//            if (files == null || files.length == 0) {
//                //noinspection ResultOfMethodCallIgnored
//                fileOrDirectory.delete();
//            }
//        }
//    }
}

File: data/src/main/java/com/github/anrimian/musicplayer/data/storage/files/FileManager.kt
package com.github.anrimian.musicplayer.data.storage.files

import java.io.File

object FileManager {

    fun deleteFile(path: String) {
        val file = File(path)
        if (!file.exists()) {
            return
        }
        file.delete()
    }

    fun deleteEmptyDirectory(fileOrDirectory: File) {
        if (fileOrDirectory.isDirectory) {
            val files = fileOrDirectory.listFiles() ?: return
            for (child in files) {
                if (fileOrDirectory.isDirectory) {
                    deleteEmptyDirectory(child)
                }
            }
            if (files.isEmpty()) {
                fileOrDirectory.delete()
            }
        }
    }
}